src/python/turicreate/toolkits/image_analysis/image_analysis.py

```diff
@@ -177,3 +190,109 @@ def resize(image, width, height, channels=None, decode=False, resample="nearest"
         raise ValueError(
             "Cannot call 'resize' on objects that are not either an Image or SArray of Images"
         )
+
+
+def get_deep_features(images, model_name, batch_size=64, verbose=True):
+    """
+    Extracts features from images from a specific model.
+
+    Parameters
+    ----------
+    images : SArray
+        Input data.
+
+    model_name : string
+        string optional
+        Uses a pretrained model to bootstrap an image classifier:
+
+           - "resnet-50" : Uses a pretrained resnet model.
+                           Exported Core ML model will be ~90M.
+
+           - "squeezenet_v1.1" : Uses a pretrained squeezenet model.
+                                 Exported Core ML model will be ~4.7M.
+
+           - "VisionFeaturePrint_Scene": Uses an OS internal feature extractor.
+                                          Only on available on iOS 12.0+,
+                                          macOS 10.14+ and tvOS 12.0+.
+                                          Exported Core ML model will be ~41K.
+
+        Models are downloaded from the internet if not available locally. Once
+        downloaded, the models are cached for future use.
+
+    Returns
+    -------
+    out : SArray
+        Returns an SArray with all the extracted features.
+
+    Examples
+    --------
+
+    """
+
+    # Check model parameter
+    allowed_models = list(_pre_trained_models.IMAGE_MODELS.keys())
+    if _mac_ver() >= (10, 14):
+        allowed_models.append("VisionFeaturePrint_Scene")
+    _tkutl._check_categorical_option_type("model", model_name, allowed_models)
+
+    # Check images parameter
+    if not isinstance(images, _tc.SArray):
+        raise TypeError("Unrecognized type for 'images'. An SArray is expected.")
+    if len(images) == 0:
+        raise _ToolkitError("Unable to extract features on an empty SArray object")
+
+    if batch_size < 1:
+        raise ValueError("'batch_size' must be greater than or equal to 1")
+
+    # Extract features
+    feature_extractor = _image_feature_extractor._create_feature_extractor(model_name)
+    images_sf = _tc.SFrame({"image":images})
+    return feature_extractor.extract_features(images_sf, "image", verbose=verbose, 
+        batch_size=batch_size)
+
+
+def find_only_image_extracted_features_column(sframe, model_name):
+    """
+    Finds the only column in `sframe` with a type of array.array and has
+    the length same as the last layer of the model in use.
+    If there are zero or more than one image columns, an exception will
+    be raised.
+    """
+    from array import array
+
+    try:
+        feature_columns, _ = zip(*list(filter(lambda x: x[1] == array, list(zip(sframe.column_names(), sframe.column_types())))))
+    except ValueError:
```

Comment:
I'm struggling to see how the above code could ever raise a `ValueError`.

Reply:
This indeed will raise a `ValueError` when the list is empty and you try to unpack the list using `zip(*...)`.
You can recreate this error with the code below:
```
import turicreate as tc
sframe_path = <Chosse a sframe with no column of type array>
sframe = tc.SFrame(sframe_path)
try :
    feature_columns, _ = zip(*list(filter(lambda x: x[1] == array, list(zip(sframe.column_names(), sframe.column_types())))))
except ValueError:
    print('It is a value error')
```